src/task_impl/mod.rs

```diff
@@ -58,7 +57,6 @@ fn with<F: FnOnce(&BorrowedTask) -> R, R>(f: F) -> R {
 pub struct Task {
     id: usize,
```

Comment:
Weird I *thought* this was deprecated but we may still be using it somewhere? Can always figure that out later of course.

Reply:
We never deprecated the actual `id` but all usage of it has been deprecated.

The `id` **should** be removed in 0.2.

Reply:
Isn't `id` still used in calls to `with_notify`, so that it will be passed down into the inner call to `notify`?

Reply:
It is indeed! That's, confusingly, a [different id though](https://github.com/rust-lang-nursery/futures-rs/blob/b11684d43be2d77e84324d309fa3e1d8b15aa076/src/task_impl/core.rs#L25), which indeed can't be removed unless we remove the ids from the `Notify` trait.

Reply:
Ah, gotcha. Removed.